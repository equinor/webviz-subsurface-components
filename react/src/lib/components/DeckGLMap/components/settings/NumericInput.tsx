import React, { ChangeEvent } from "react";
import { Label, Input } from "@equinor/eds-core-react";

interface Props {
    /**
     * Label for the component.
     */
    label: string;
    /**
     * Initial state of the component.
     */
    value: number;
    /**
     * Callback to update the state of the component.
     */
    onChange: (e: ChangeEvent<HTMLInputElement>) => void;
}

const NumericInput: React.FC<Props> = React.memo(
    ({ label, value, onChange }: Props) => {
        return (
            <div
                style={{
                    display: "flex",
                    justifyContent: "space-between",
                }}
            >
                <Label
                    label={label}
                    id={`${label}-input-label`}
                    style={{
                        paddingTop: 5,
                        paddingBottom: 5,
                        fontSize: 15,
                    }}
                />
                <Input
                    id={`${label}-input`}
                    type={"number"}
                    value={value}
                    onChange={onChange}
                    min={0}
                    style={{
                        fontSize: 15,
                        textAlign: "right",
                        width: "3rem",
                    }}
                />
            </div>
        );
    }
);

NumericInput.displayName = "NumericInput";
export default NumericInput;
