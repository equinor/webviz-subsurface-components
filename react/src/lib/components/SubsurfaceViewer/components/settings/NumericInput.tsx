import React, { ChangeEvent } from "react";
import { Label, Input } from "@equinor/eds-core-react";

interface Props {
    /**
     * Label for the component.
     */
    label: string;
    /**
     * Initial state of the component.
     */
    value: number;

    min?: number;
    max?: number;
    step?: number;

    /**
     * Callback to update the state of the component.
     */
    onChange: (e: ChangeEvent<HTMLInputElement>) => void;
}

const NumericInput: React.FC<Props> = React.memo(
    ({ label, value, min, max, step, onChange }: Props) => {
        return (
            <div
                style={{
                    display: "flex",
                    justifyContent: "space-between",
                }}
            >
                <Label
                    label={label}
                    id={`${label}-input-label`}
                    style={{
                        paddingTop: 5,
                        paddingBottom: 5,
                        fontSize: 15,
                    }}
                />
                <Input
                    id={`${label}-input`}
                    type={"number"}
                    value={value}
                    onChange={onChange}
                    min={min}
                    max={max}
                    step={step}
                    style={{
                        fontSize: 15,
                        textAlign: "right",
                        width: "3rem",
                    }}
                />
            </div>
        );
    }
);

NumericInput.defaultProps = {
    min: 0,
    step: 1,
};

NumericInput.displayName = "NumericInput";
export default NumericInput;
