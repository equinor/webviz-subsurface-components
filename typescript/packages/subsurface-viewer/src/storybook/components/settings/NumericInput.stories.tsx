import type { Meta, StoryObj } from "@storybook/react";

import NumericInput from "../../../components/settings/NumericInput";

const stories: Meta = {
    component: NumericInput,
    title: "SubsurfaceViewer/Components/Settings",
};
export default stories;

export const NumericInputStory: StoryObj<typeof NumericInput> = {
    name: "NumericInput",
    args: {
        label: "test",
        value: 5,
    },
};
