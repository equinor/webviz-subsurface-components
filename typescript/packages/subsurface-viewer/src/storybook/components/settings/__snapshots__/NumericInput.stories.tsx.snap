// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubsurfaceViewer/Components/Settings NumericInputStory smoke-test 1`] = `
<div style="display: flex; justify-content: space-between;">
  <label id="test-input-label"
         class="Label__LabelBase-sc-1gi2bcn-0 jGGvso"
         style="padding-top: 5px; padding-bottom: 5px; font-size: 15px;"
  >
    <span class="Label__Text-sc-1gi2bcn-1 bjgoJD">
      test
    </span>
  </label>
  <div class="Input__Container-sc-1ykv024-0 efGlqQ"
       style="font-size: 15px; text-align: right; width: 3rem;"
  >
    <input type="number"
           id="test-input"
           min="0"
           step="1"
           class="Input__StyledInput-sc-1ykv024-1 jtVHwP"
           value="5"
           style="resize: none;"
    >
  </div>
</div>
`;
